pump/storage/vlog_linux_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package storage
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/docker/docker/pkg/mount"
+	"github.com/pingcap/check"
+)
+
+// Need root privilege for using mount to get a specify tmpfs file system
```

Comment:
IMO this is very heavy for needing `root` to run this test, and even worse binlog now depends on the entire Docker repository 🙃.

I suggest we simply set `RLIMIT_FSIZE` to a small number to ensure file size cannot increase more than 40 KB.

```go
package main

import (
	"syscall"
	"fmt"
	"os"
)

func main() {
	var origRlimit syscall.Rlimit
	err := syscall.Getrlimit(syscall.RLIMIT_FSIZE, &origRlimit)
	if err != nil { panic(err) }

	err = syscall.Setrlimit(syscall.RLIMIT_FSIZE, &syscall.Rlimit{Cur: 40, Max: origRlimit.Max})
	if err != nil { panic(err) }

	f, err := os.Create("test-fsize-rlimit.txt")
	if err != nil { panic(err) }

	defer f.Close()

	for i := 0; i < 5; i++ {
		n, err := f.Write([]byte("test content\n"))
		fmt.Println(n, err)
	}

	err = syscall.Setrlimit(syscall.RLIMIT_FSIZE, &origRlimit)
	if err != nil { panic(err) }

	for i := 0; i < 5; i++ {
		n, err := f.Write([]byte("test content\n"))
		fmt.Println(n, err)
	}
}
```

Reply:
good idea
change to use `RLIMIT_FSIZE ` https://github.com/pingcap/tidb-binlog/pull/640/commits/6683f2c68f9eada5e48e18447c58f73ab925f8fc